composefs.spec.in

```diff
@@ -22,6 +22,7 @@ Devel files for %{name}.
 
 %package        libs
 Summary:        Libraries for %{name}
+License:        LGPL-3.0-or-later AND Apache-2.0
```

Comment:
I think this isn't quite right. All our code in the library should be lgpl3-or-later, except the erofs_fs.h header, and that one is GPL2-only OR Apacahe-2.0).

I believe this is important, as it makes the library compatible with GPLv2 only apps. (GPL2 is incompat with apache2, but gpl3 is not.

Reply:
this came up from the tools looking at `composefs/libcomposefs/lcfs-fsverity.c` and me not reading the code directly:

```
detected_license_expression_spdx: LGPL-2.1-or-later AND LGPL-3.0-or-later
```

looking at line 53 which is what it said was LGPL-3.0 it actually says:
```
/* adapted from the SHA256 implementation in glib, which is originally:
 *
 * Copyright (C) 2006 Dave Benson
 * Released under the terms of the GNU Lesser General Public License
 */

```

Will fix that entry

Reply:
'fixed' the entry, and covered the other license issue I got in the fedora-review tool.

Reply:
> I think this isn't quite right. All our code in the library should be lgpl3-or-later, except the erofs_fs.h header, and that one is GPL2-only OR Apacahe-2.0).
> 
> I believe this is important, as it makes the library compatible with GPLv2 only apps. (GPL2 is incompat with apache2, but gpl3 is not.

I could switch to  GPL2-or-later OR Apache-2.0 for the header leter if needed.

Reply:
@hsiangkao I think the current licensing for it is ok, There are some combinations:

App is apache2: use apache2 for erofs_fs.h and lgpl3 for libcomposefs
App is GPLv3: use apache2 for erofs_fs.h and lgpl3 for libcomposefs
App is GPLv2 only: Use gplv2 for erofs_fs.h and lgpl3 for libcomposefs
App is GPLv2 or later, chose one of the above

All of these avoid the GPLv2 and apache combination that is problematic.

However, if you ever was interested in changing the license of the header I would propose something very unlimited, like MIT.

Reply:
I think we can consider this later (to MIT) since it's quite a serious switch (compared with GPLv2-only to GPLv2-or-later).
If there are some hard requirements we'd like to integrate later, we could switch to MIT then.

Anyway, a lot of cloud native softwares and AOSP itself use Apache 2.0 license.